test/tests.js

```diff
@@ -123,253 +131,353 @@ describe('Optimizely Forwarder', function () {
+                "variables": [],
+                "featureFlags": [
+                  {
+                    "experimentIds": [],
+                    "rolloutId": "18375930463",
+                    "variables": [
+                      {
+                        "defaultValue": "",
+                        "type": "string",
+                        "id": "18375860245",
+                        "key": "f2variable1"
+                      }
+                    ],
+                    "id": "18360310384",
+                    "key": "featurename2"
+                  },
+                  {
+                    "experimentIds": [],
+                    "rolloutId": "18389000125",
+                    "variables": [
+                      {
+                        "defaultValue": "",
+                        "type": "string",
+                        "id": "18358530123",
+                        "key": "f1variable1"
+                      }
+                    ],
+                    "id": "18381870404",
+                    "key": "featurename1"
+                  }
+                ],
+                "experiments": [],
+                "accountId": "7015510640",
+                "events": [
+                  {
+                    "experimentIds": [],
+                    "id": "18356890341",
+                    "key": "eventKey2"
+                  },
+                  {
+                    "experimentIds": [],
+                    "id": "18391220198",
+                    "key": "eventKey1"
+                  }
+                ]
+              }
+            // Include any specific settings that are required for initializing your SDK here
+            var sdkSettings = {
+                projectId: 'LYLgZJqZzFKd5SaNLcQRc'
+            };
+            // You may require userAttributes or userIdentities to be passed into initialization
+            var userAttributes = {
+                color: 'green'
+            };
+            var userIdentities = [{
+                Identity: 'customerId',
+                Type: IdentityType.CustomerId
+            }, {
+                Identity: 'email',
+                Type: IdentityType.Email
+            }, {
+                Identity: 'facebook',
+                Type: IdentityType.Facebook
+            }];
+
+            mParticle.forwarder.init(sdkSettings, reportService.cb, true, null, userAttributes, userIdentities);
+        });
+
+        it('should log a custom event', function(done) {
+            mParticle.forwarder.process({
+                EventDataType: MessageType.PageEvent,
+                EventName: 'eventKey1',
+                UserId: 'tkatz@mparticle.com',
+                EventAttributes: {
+                    label: 'label',
+                    value: 200,
+                    category: 'category'
+                },
+                CustomFlags: {
+
+                }
+            });
+
+            window.optimizelyDatafile.events.length.should.equal(2);       
```

Comment:
i would take a look at each of the events to make sure they look like what you expect them to, and not just test the event length.  

was there a reason that there are 2 events here and not just 1?

let's pair together on adding a few more tests:
1. events that are no in the optimizely test should not be tracked
2. ensure events are properly tracking user attributes / user identities (test for a set userIdentity like `customerId`, and then test for if a user identity is null, to make sure it's being set to the `deviceId`)
3. the above for each of commerce and regular events